src/raftstore/store/fsm/batch.rs

```diff
@@ -68,52 +68,33 @@ macro_rules! impl_sched {
                 }
             }
         }
-    };
-}
-
-impl_sched!(NormalScheduler);
-impl_sched!(ControlScheduler);
 
-impl<N: Fsm, C> FsmScheduler for NormalScheduler<N, C> {
-    type Fsm = N;
+        impl<N: Fsm, C: Fsm> FsmScheduler for $name<N, C> {
+            type Fsm = $fsm;
 
-    #[inline]
-    fn schedule(&self, fsm: Box<N>) {
-        match self.sender.send(FsmTypes::Normal(fsm)) {
-            Ok(()) => return,
-            // TODO: use debug instead.
-            Err(SendError(FsmTypes::Normal(fsm))) => warn!("failed to schedule fsm {:p}", fsm),
-            _ => unreachable!(),
-        }
-    }
+            #[inline]
+            fn schedule(&self, fsm: Box<Self::Fsm>) {
+                match self.sender.send($ty(fsm)) {
+                    Ok(()) => return,
+                    // TODO: use debug instead.
+                    Err(SendError($ty(fsm))) => warn!("failed to schedule fsm {:p}", fsm),
+                    _ => unreachable!(),
+                }
+            }
 
-    fn shutdown(&self) {
-        // TODO: close it explicitly once it's supported.
-        // Magic number, actually any number greater than poll pool size works.
-        for _ in 0..100 {
-            let _ = self.sender.send(FsmTypes::Empty);
+            fn shutdown(&self) {
+                // TODO: close it explicitly once it's supported.
+                // Magic number, actually any number greater than poll pool size works.
+                for _ in 0..100 {
+                    let _ = self.sender.send(FsmTypes::Empty);
+                }
+            }
         }
-    }
+    };
```

Comment:
Why the semicolon? 😅

Reply:
This semicolon comes from L71 in the previous code.